clang/docs/ReleaseNotes.rst

```diff
@@ -203,6 +203,10 @@ Non-comprehensive list of changes in this release
 - ``__typeof_unqual__`` is available in all C modes as an extension, which behaves
   like ``typeof_unqual`` from C23, similar to ``__typeof__`` and ``typeof``.
 
+- Improved stack usage with C++ initialization code. This allows significantly
+  more template instantiations before reaching stack exhaustion limits.
```

Comment:
```suggestion
  more levels of recursive initialization before reaching stack exhaustion limits.
```

This isn't REALLY a template problem, and you could have manually created this problem, but templates make it 'easier' to hit this.  Perhaps the commit message should be improved too?